File: src/main/java/github/macro/build_info/gems/acquisition/IngredientType.java
package github.macro.build_info.gems.acquisition;

import org.jetbrains.annotations.Nullable;

import java.util.Arrays;

/**
 * Created by Macro303 on 2020-Jul-14.
 */
public enum IngredientType {
	DIVINATION_CARD,
	GEM,
	CURRENCY,
	EQUIPMENT;

	@Nullable
	public static IngredientType value(String name) {
		return Arrays.stream(values()).filter(tag -> tag.name().replaceAll("_", " ").equalsIgnoreCase(name)).findFirst().orElse(null);
	}
}

File: src/main/kotlin/github/macro/build_info/gems/ingredient/IngredientType.kt
package github.macro.build_info.gems.ingredient

/**
 * Created by Macro303 on 2020-Feb-27.
 */
enum class IngredientType {
	DIVINATION_CARD,
	GEM,
	CURRENCY,
	EQUIPMENT;

	companion object {
		fun value(name: String): IngredientType? = values().firstOrNull {
			it.name.replace("_", " ").equals(name, ignoreCase = true)
		}
	}
}